### PR TITLE
Fixes #29332 - update the breadcrumbs to pf4

### PR DESCRIPTION
**Pull request by sharvit**

### CODE REVIEW

**Comment by MariaAga** [comment]
Nice change! It seems that if the title is too long it doesnt get cut like it used to
image

image

**Comment by sharvit** [comment]
Thanks @MariaAga, fixed

**Review by MariaAga — DISMISSED** [review]

**Comment by amirfefer** [comment]
Thanks @sharvit ! :+1: 
pf3
FireShot Capture 226 - lizalaptop2 tlv redhat com - localhost
pf4
FireShot Capture 228 - lizalaptop2 tlv redhat com - localhost

looks like pf4 uses a different font and much smaller size, still looks great.
the operating system logo alignment has been broken a bit though

**Comment by sharvit** [comment]
> Thanks @sharvit ! 👍
> pf3
> FireShot Capture 226 - lizalaptop2 tlv redhat com - localhost
> pf4
> FireShot Capture 228 - lizalaptop2 tlv redhat com - localhost
> 
> looks like pf4 uses a different font and much smaller size, still looks great.
> the operating system logo alignment has been broken a bit though

Actually it's the same font but the sizes are meant to use with a different font, It would look much better (font, sizing, icon) once https://github.com/theforeman/foreman/pull/7519 would merge. See: https://user-images.githubusercontent.com/1262502/76521884-5694e480-646e-11ea-8e62-87b8e02a421e.png

I'm wondering if I should put some patch until #7519 will get in...

**Comment by sharvit** [comment]
Rebased. Using the pf4 fonts now so it should look much better:
Screen Shot 2020-06-22 at 15 04 51

File: webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/__snapshots__/Breadcrumb.test.js.snap [hunk 1]
```diff
@@ -1,31 +1,29 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`Breadcrumbs renders breadcrumbs menu 1`] = `
-<Breadcrumb
-  title={false}
->
+<Component>
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/__snapshots__/Breadcrumb.test.js.snap]: is this expected?

Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/__snapshots__/Breadcrumb.test.js.snap]: Based on https://enzymejs.github.io/enzyme/docs/api/ReactWrapper/name.html

> The order of precedence on returning the name is: `type.displayName` -> `type.name` -> `type`.

I assume that PF4 compiled it into an anonymous function and didn't set a `displayName` or `name`.

Comment (ezr-ondrej) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/__snapshots__/Breadcrumb.test.js.snap]: Should we set the `displayName` manually?

Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/__snapshots__/Breadcrumb.test.js.snap]: I don’t think we should as it's a 3rd party component, as far as we keep our snapshots short, we should be fine.

Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/BreadcrumbBar/components/__snapshots__/Breadcrumb.test.js.snap]: PF4 fixed the issue with the display name: https://github.com/patternfly/patternfly-react/pull/4391

This PR in foreman-js should fix that issue: https://github.com/theforeman/foreman-js/pull/146

**Comment by sharvit** [comment]
Rebased, snapshots should be back to normal after pf4 fixed the display-name issue.

**Comment by ezr-ondrej** [comment]
The text should be above the line shouldn't it?
BreadcrumsPF4
Or am I doing something wrong? I've reinstalled node_modules, but didn't help.

**Comment by MariaAga** [comment]
Works for me
Screenshot from 2020-07-19 10-47-13
@ezr-ondrej maybe it's still using the deleted css? I've added it back and it gave your result.
Try running `rake assets:clobber`

**Review by ezr-ondrej — APPROVED** [review]
:facepalm: You're right @MariaAga, I was using the old css.
Thank you @sharvit and @MariaAga for driving the review! :+1:

**Pull request merged**